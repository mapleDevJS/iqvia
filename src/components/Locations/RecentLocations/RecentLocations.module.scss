.recentLocations {
    margin: 0;
    padding: 0;
}

.location {
    display: flex;
    align-items: center;
    margin: 0;
    padding: 0.25rem 0;
    list-style: none;
    border-bottom: 1px solid var(--border-light);

    @media (max-width: 412px) {
        flex-wrap: wrap;
    }
}

.locationName {
    text-align: left;

    @media (max-width: 412px) {
        width: 100%;
    }
}

.locationLink {
    padding: 0.25rem 0.25rem 0.25rem 0;
    font-weight: 600;
    text-align: start;
    background-color: transparent;
    border-style: none;

    &:hover {
        color: var(--font-alt);
        cursor: pointer;
    }
}

.weatherIcon {
    flex-shrink: 0;
    width: 48px;
    height: 48px;
}

.buttonGroup {
    flex-shrink: 0;
    width: 5rem;
    margin-left: auto;
}
