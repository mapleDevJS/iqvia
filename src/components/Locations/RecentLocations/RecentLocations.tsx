import { useCallback, useContext } from 'react';
import { store } from 'store/index';
import { observer } from 'mobx-react';
import { debounce } from 'lodash';
import { Location } from 'types/index';
import commonStyles from 'styles/common.module.scss';
import styles from './RecentLocations.module.scss';
import { ReactComponent as RefreshIcon } from 'assets/icons/refresh.svg';
import { ReactComponent as DeleteIcon } from 'assets/icons/delete.svg';
import { CELSIUS } from 'consts';

interface Props {
    locations: Location[];
}

const RecentLocations: React.FC<Props> = ({ locations }) => {
    const { locationsStore } = useContext(store);
    const { fetchLocationById, fetchForecast, deleteLocationById, resetForecast, getForecast } = locationsStore;

    const forecast = getForecast();

    // eslint-disable-next-line react-hooks/exhaustive-deps
    const debouncedFetchLocationById = useCallback(debounce(fetchLocationById, 400), []);
    // eslint-disable-next-line react-hooks/exhaustive-deps
    const debouncedFetchForecast = useCallback(debounce(fetchForecast, 400), []);

    return (
        <ul className={styles.recentLocations}>
            {locations.map(location => (
                <li key={location.id} className={styles.location}>
                    <div className={styles.locationName}>
                        <button className={styles.locationLink} onClick={() => debouncedFetchForecast(location.id)}>
                            {location.name}
                        </button>
                    </div>
                    <span>
                        {location.main.temp}&nbsp;{CELSIUS}
                    </span>

                    <span className={styles.weatherIcon}>
                        <img
                            src={require(`assets/icons/${location.weather[0].icon}.svg`).default}
                            alt={location.weather[0].description}
                            width="48"
                            height="48"
                        />
                    </span>
                    <span>{location.weather[0].main}</span>

                    <div className={styles.buttonGroup}>
                        <button
                            className={commonStyles.refreshButton}
                            onClick={() => debouncedFetchLocationById(location.id)}
                        >
                            <RefreshIcon />
                        </button>
                        <button
                            className={commonStyles.refreshButton}
                            onClick={() => {
                                if (forecast?.city.id === location.id) resetForecast();
                                deleteLocationById(location.id);
                            }}
                            data-testid="delete-button"
                        >
                            <DeleteIcon />
                        </button>
                    </div>
                </li>
            ))}
        </ul>
    );
};

export default observer(RecentLocations);
