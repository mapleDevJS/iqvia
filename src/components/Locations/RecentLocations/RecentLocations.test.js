import { render, screen, cleanup } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import RecentLocations from './RecentLocations';
import { ReactComponent as RefreshIcon } from 'assets/icons/refresh.svg';

afterEach(cleanup);

describe('when rendered', () => {
    test('renders RecentLocations component', () => {
        render(<RecentLocations locations={[]} />);
    });
});

describe('when add button clicked', () => {
    it('calls the onClick callback handler', () => {
        const onClick = jest.fn();

        render(
            <button
                className=""
                onClick={onClick}
            >
                <RefreshIcon />
            </button>
        );

        userEvent.click(screen.getByRole('button'));
        expect(onClick).toHaveBeenCalledTimes(1);
    });
});

describe('snapshot', () => {
    it('should take a snapshot', () => {
        const { asFragment } = render(<RecentLocations locations={[]}/>)
        expect(asFragment(<RecentLocations locations={[]}/>)).toMatchSnapshot()
    })
});
