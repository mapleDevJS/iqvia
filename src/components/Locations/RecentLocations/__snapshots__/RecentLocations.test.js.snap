// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot should take a snapshot 1`] = `
<DocumentFragment>
  <ul
    class="recentLocations"
  />
</DocumentFragment>
`;
